sdks/go/pkg/beam/runners/prism/internal/jobservices/server_test.go

```diff
@@ -121,6 +132,30 @@ func TestServer_RunThenCancel(t *testing.T) {
 		t.Fatalf("server.Run() = returned empty preparation ID, want non-empty")
 	}
 
+	// Wait for the job to be in the RUNNING state before we cancel it.
+	const (
+		maxRetries = 10
+		retrySleep = 100 * time.Millisecond
+	)
+	var jobIsRunning bool
+	for range maxRetries {
+		stateResp, err := undertest.GetState(ctx, &jobpb.GetJobStateRequest{JobId: runResp.GetJobId()})
```

Comment:
I'd use the streaming version of the call, not the one-off . Then there's less dynamic waiting for things to happen for up to a second.